### ISSUE TITLE
[A11y] 'Find in page' feature is hard to use with TalkBack

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/focus-android#2948]
## Steps to reproduce
1. Activate TalkBack.
2. Open a webpage.
3. Open the main menu and select 'Find in page'.
4. Try to type something.

### Expected behavior
When the search bar is opened, there should be a signal that it's displayed and if the keyboard is on/off.
The keyboard should be on by default.

### Actual behavior
When the search bar is opened, there is no signal that it's opened, or if the keyboard is on/off.
The keyboard is hidden.

### Device information
* Android device: 
LG Nexus 5 (Android 6.0.1)
Google Pixel (Android P beta)
* Focus version: 6.3

**Comment by vesta0** [comment]
@sv-ohorvath please test to confirm this is still an issue.

**Comment by colintheshots** [comment]
The current Talkback experience on launching find in page is this: "Editing find in page edit box. Showing English QWERTY keyboard. Firefox Focus." It then proceeds to read off each letter or digit I enter.

I noticed that I lose the URL bar at the top if I close the find in page toolbar at the bottom, so it's not easy to start searching again.

When I press next and previous, I hear a brief tap sound like a tom-tom drum. The actual search experience is not easy, but it's not clear how best it should function in TalkBack mode. Should we read the surrounding page context?

**Comment by brampitoyo** [comment]
@ekager @colintheshots Thanks for doing all your research on this!

I’m not an expert on Talkback, but I think announcing that the text box is ready for input is a good idea.

As Emily had suggested, the string can be:

> Find in page. To enter a text in this field, type

Should we announce each letter or digit that the user enters? I think so. This is the Android system behaviour when Talkback is activated, right?

-----

Let’s say that the page contains this text (it’s the GitHub homepage):

> **Built for developers**
> 
> GitHub is a development platform inspired by the way you work. From [open source]() to [business](), you can host and review code, manage projects, and build software alongside 28 million developers.

I type “b”. Talkback announces:

> B. 253 matches found.
> Match number 1. Heading: Built for developers.

I type “u”. Talkback announces:

> U. Bu. 3 matches found.
> Match number 1. Heading: Built for developers.

Notice that talkback says the letter, then tries to say the sound unit that’s produced after typing “B” and “U”, which is “bu”.

Now I press the “Next result” button. Talkback announces:

> Match number 2. From link: open source – to link: business.

Notice that Talkback stops at the comma, but reads the content of the phrase.

I press “Next result” again. Talkback announces:

> Match number 3. And build software alongside 28 million developers.
> End of match.

Do we need to announce that the next and previous results buttons are available? I don’t think so. The fact that there are more than one matches implies that the user can press the “Next result” button.

Does this sound reasonable? (Can we control Talkback to this extent?)

**Comment by ekager** [comment]
I definitely think this is ideal behavior, but I'm not 100% sure if we can currently precisely activate the WV/GV like that to read the matches.

**Comment by colintheshots** [comment]
How do we decide how much context text to provide for each result? Should we read the current sentence or para?

**Comment by ekager** [comment]
If you use a browser (WV or GV I believe) with TalkBack and click on a section of the website, it will read a pre-determined section which I'm assuming has to do with the website's formatting. I don't think we would have to decide this?

**Comment by brampitoyo** [comment]
@ekager @colintheshots It would be good to rely on TalkBack to read the correct context. This way, we won’t have to cover every single corner cases in reading websites and UI texts!

**Comment by MarcoZehe** [comment]
Yes, the context may vary greatly, like a link, some text in a paragraph etc. Letting TalkBack decide upon focus is probably a good way forward. Also, we can iterate on this if necessary. Like get a reasonable implementation done first, and then collect user feedback and improve on it if that contains actionable items.

**Comment by eeejay** [comment]
From looking at how Chrome does this, it is similar to @brampitoyo's suggestion above. Didn't dig deep, but they may be implementing the result count speech output by using a [TYPE_ANNOUNCEMENT](https://developer.android.com/reference/android/view/accessibility/AccessibilityEvent.html#TYPE_ANNOUNCEMENT) . Then when the user presses next/previous, somehow accessibility focus moves to where the DOM selection is (doesn't yet work on GV, didn't test on WK).
